File: chip8/src/display.ts
class Display {
    // Framebuffer should be written to memory at addresses 0xF00 - 0xFFF
    framebuffer: u8[] = new Array(64 * 32);

    draw (x: u8, y: u8, on: bool): void {
        this.framebuffer[x + y * 64] = on ? 1 : 0;
    }

    clear (): void {
        this.framebuffer.fill(0);
    }
}

export const display = new Display();

File: src/display.rs
pub struct Display {
    // Framebuffer should be written to memory at addresses 0xF00 - 0xFFF
    framebuffer: [u8; 64 * 32],
}

impl Display {
    fn draw (&mut self, x: u8, y: u8, on: bool) {
        self.framebuffer[(x + y * 64) as usize] = if (on) { 1 } else { 0 };
    }

    fn clear (&mut self) {
        self.framebuffer = [0; 64 * 32];
    }
}

pub static DISPLAY: Display = Display {
    framebuffer: [0; 64 * 32], 
};
